pkg/volume/azure_dd/azure_provision.go

```diff
@@ -175,6 +180,16 @@ func (p *azureDiskProvisioner) Provision(selectedNode *v1.Node, allowedTopologie
 		return nil, err
 	}
 
+	if kind != v1.AzureManagedDisk {
+		if resourceGroup != "" {
+			return nil, errors.New("StorageClass option 'resourceGroup' can be used only for managed disks")
+		}
+
+		if zoned {
```

Comment:
`zoned` seems to only have been initialized above this and assigned to down below through the call to `parseZoned`. Should this check be done after `zoned` is populated through `parseZoned`?

Reply:
Oops, you're right.  Let me file another PR fix this